### PR TITLE
Update events

### PR DESCRIPTION
**Pull request by ben-kaufman**
Adds and updates some events to improve the subgraph

### CODE REVIEW

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 1]
```diff
@@ -163,6 +163,33 @@ interface IHATVaultsRegistry {
      */
     error NotEnoughFeePaid();
 
+    /**
+     * @notice Emitted on deployment of the registry
+     * @param _hatVaultImplementation The HATVault implementation address
+     * @param _HAT the HAT token address
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:169]:
```suggestion
     * @param _HAT The HAT token address
```

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 2]
```diff
@@ -163,6 +163,33 @@ interface IHATVaultsRegistry {
...
+     * @param _tokenLockFactory the token lock factory address
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:170]:
```suggestion
     * @param _tokenLockFactory The token lock factory address
```

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 3]
```diff
@@ -163,6 +163,33 @@ interface IHATVaultsRegistry {
...
+     * @param _generalParameters the registry's general parameters
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:171]:
```suggestion
     * @param _generalParameters The registry's general parameters
```

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 4]
```diff
@@ -163,6 +163,33 @@ interface IHATVaultsRegistry {
...
+     * @param _bountyGovernanceHAT The HAT bounty for governance
+     * @param _bountyHackerHATVested The HAT bounty vested for the hacker
+     * @param _hatGovernance The registry's governance
+     * @param _defaultChallengePeriod The new default challenge period
+     * @param _defaultChallengeTimeOutPeriod The new default challenge timeout
+     * @param _defaultArbitratorCanChangeBounty Whether the arbitrator can change bounty of claims
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:177]:
```suggestion
     * @param _defaultArbitratorCanChangeBounty Whether the arbitrator can change bounty percentage of claims
```

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 5]
```diff
@@ -163,6 +163,33 @@ interface IHATVaultsRegistry {
...
+     * describes the claim.
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:178]:
```suggestion
```

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 6]
```diff
@@ -248,6 +275,7 @@ interface IHATVaultsRegistry {
     event VaultCreated(
         address indexed _vault,
         address indexed _asset,
+        address indexed _owner,
```
Comment (lirona) [inline contracts/interfaces/IHATVaultsRegistry.sol:278]: add to documentation

**Review by lirona — COMMENTED** [review]
looks great, maybe add the new event to the tests? i think we don't have a test for registry creation so maybe nevermind..

**Review by jellegerbrandy — COMMENTED** [review]
This looks good to me in principal. 
I am just not sure about the events for RegistryCreated and RewardControllerCreated. These contracts will be deployed only once, and i do not think our subgraph implementation will ever need to listen to these events

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**